models/fact.js

```diff
@@ -271,6 +272,113 @@ function getFact({ number, type, options = {} }) {
     });
   }
 }
+/**
+ * This function mutates the given empty object to add keys for each type
+ * This is a helper function for getAllFacts and getTypeFacts
+ *
+ * start {}
+ * ends {type: [], type:[]}
+ *
+ * @param types: string[],
+ * @param object: empty object
+ * @param dateNum: number converted to conform to date number
+ * @param num: current number
+ *
+ *
+ */
+function getFactTexts(types, object, dateNum, num) {
+  for (let type of types) {
+    if (type === "date" && data[type][dateNum]) {
+      let prefix = utils.getStandalonePrefix(dateNum, type);
+      object[type] = data[type][dateNum].map(({ text }) => `${prefix} ${text}`);
+    } else if (data[type][num]) {
+      object[type] = data[type][num].map(({ text }) => text);
+    } else {
+      object[type] = [getDefaultMsg({ number: num, type })];
+    }
+  }
+}
+
+/**
+ * getAllFacts() is used as a graphQL endpoint in lieu of querying a database
+ *
+ * @param num: is an int
+ * @returns {
+ *            number: num
+ *            year: string[],
+ *            trivia: string[],
+ *            math: string[],
+ *            date: string[]
+ *          }
+ * The return must conform to the GraphQL NumberType expected fields, which are currently {number, year, trivia, math, date}
+ */
+function getAllFacts(num) {
+  let res = {};
+  let types = ["year", "trivia", "math", "date"];
+  let dateNum = utils.dateToDayOfYear(new Date(2004, 0, num));
+
+  getFactTexts(types, res, dateNum, num);
```

Comment:
Not a big fan of passing res to the helper function by reference.
I think it is easier to reason about if we do something like this instead.
```
let res = getFactTexts(types, dateNum, num);
res.number = num;
```

Reply:
Updated all instances of getFactTexts to reflect this change